HW4_smythe_full.py

```diff
@@ -0,0 +1,370 @@
+"""
+Created by Ed Smythe Oct 2020
+BDA 696 @ SDSU
+Given a pandas dataframe
+    Contains both a response and predictors
+Given a list of predictors and the response columns
+    Determine if response is continuous or boolean
+    (don't worry about >2 category responses)
+    Loop through each predictor column
+        Determine if the predictor cat/cont
+        Automatically generate the necessary plot(s) to inspect it
+        Calculate the different ranking algos
+        p-value & t-score along with it's plot
+        Regression: Continuous response
+        Logistic Regression: Boolean response
+        Difference with mean of response along with it's plot (weighted and unweighted)
+        Random Forest Variable importance ranking
+        Generate a table with all the variables and their rankings
+I'm going to grade this by giving it some random dataset
+and seeing if it outputs everything
+Desire: html or excel based rankings report with links to the plots
+
+    Recommendations for plots
+    Predictor / Response type Dependant
+        Response: Boolean / Categorical
+            Predictor: Boolean / Categorical
+                Heatplot
+            Predictor: Continous
+                Violin plot on predictor grouped by response
+                Distribution plot on predictor grouped by response
+        Response: Continuous
+            Predictor: Boolean / Categorical
+                Violin plot on response grouped by predictor
+                Distribution plot on response grouped by predictor
+            Predictor: Continuous
+                Scatter plot with trendline
+"""
+
+import sys
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import statsmodels.api as sm
+from sklearn import datasets
+from sklearn.linear_model import LogisticRegression

```

Comment:
I don't think we need LogisiticRegression. I think we just needed RandomForestClassifier, RandomForestRegressor.